features/admin.server-configurations.v1/components/__tests__/password-recovery-form.test.tsx

```diff
@@ -0,0 +1,170 @@
+                "name": "Recovery.Notification.Password.ExpiryTime.smsOtp",
+                "value": "1"
+            },
+            {
+                "description": "",
+                "displayName": "Max failed attempts for password recovery",
+                "name": "Recovery.Notification.Password.MaxFailedAttempts",
+                "value": "3"
+            },
+            {
+                "description": "",
+                "displayName": "Max resend attempts for password recovery",
+                "name": "Recovery.Notification.Password.MaxResendAttempts",
+                "value": "5"
+            },
+            {
+                "description": "",
+                "displayName": "Notification based password recovery via an email",
+                "name": "Recovery.Notification.Password.emailLink.Enable",
+                "value": "false"
+            },
+            {
+                "description": "",
+                "displayName": "Notification based password recovery using SMS OTP",
+                "name": "Recovery.Notification.Password.smsOtp.Enable",
+                "value": "false"
+            }
+        ],
+        "subCategory": "DEFAULT"
+    },
+    isConnectorEnabled: true,
+    onSubmit: onSubmitMock
+};
+
+describe("PasswordRecoveryForm", () => {
+
+    it("renders the PasswordRecoveryForm component.", async () => {
+        render(<PasswordRecoveryConfigurationForm
+            { ...initialProps }
+            data-componentid="password-recovery-edit-form" />);
+
+        expect(screen.getByTestId(`${data_component_id}-email-link-based-recovery`)).toBeInTheDocument();
+    });
+
+    it("recovery option selection checkboxes are checkable.", async () => {
+        render(<PasswordRecoveryConfigurationForm { ...initialProps } />);
+
+        const email_checkbox: HTMLInputElement = screen.getByTestId(`${data_component_id}-email-link-based-recovery`);
+        const sms_checkbox: HTMLInputElement = screen.getByTestId(`${data_component_id}-sms-based-recovery`);
+
+        fireEvent.click(email_checkbox);
+        fireEvent.click(sms_checkbox);
+
+        expect(email_checkbox).toBeChecked();
+        expect(sms_checkbox).toBeChecked();
+    });
+
+    it("otp code configuration checkboxes don't allow all checkboxes to be unchecked.", async () => {
+        render(<PasswordRecoveryConfigurationForm { ...initialProps } />);
+
+        const uppercase_checkbox: HTMLInputElement = screen.getByTestId(`${data_component_id}-sms-otp-uppercase`);
+        const lowercase_checkbox: HTMLInputElement = screen.getByTestId(`${data_component_id}-sms-otp-lowercase`);
+        const numeric_checkbox: HTMLInputElement = screen.getByTestId(`${data_component_id}-sms-otp-numeric`);
+
+        (!uppercase_checkbox.checked) && fireEvent.click(uppercase_checkbox);
+        (!lowercase_checkbox.checked) && fireEvent.click(lowercase_checkbox);
+        (numeric_checkbox.checked) && fireEvent.click(numeric_checkbox);
```

Comment:
Let's use the if else syntax for better readability.

Reply:
Addressed with https://github.com/wso2/identity-apps/pull/6418/commits/b42935fe7590aab3d67eae574a7613217772d25e